### PR TITLE
Sample 4: Using memory and embeddings with GitHub

### PR DESCRIPTION
**Pull request by craigomatic**
### Motivation and Context
This sample continues in the spirit of prior samples in the samples/apps folder with a goal of teaching developers how the concepts of memory and embeddings can be integrated.

### Description
This sample includes a frontend app that allows the developer to enter the address of a publicly available Github repository, which is downloaded to the temp directory of the machine hosting the KernelHttpService sample application (can be the same machine).

image

Once downloaded the developer is able to have a conversation with the repository and see the impact of the relevance slider on the responses. 

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/github-qna-webapp-react/src/components/GitHubRepoSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    useEffect(() => {
+        if (isSameProjectAndBranch()) {
+            setDownloadState(DownloadState.Loaded);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [branch]);
+
+    return (
+        <div style={{ paddingTop: 20, gap: 20, display: 'flex', flexDirection: 'column', alignItems: 'left' }}>
+            <Title3 style={{ alignItems: 'left' }}>Enter in the GitHub Project URL</Title3>
```
Comment (shawncal) [inline samples/apps/github-qna-webapp-react/src/components/GitHubRepoSelection.tsx:74]: "Enter in" sounds weird to me...

Recommend either "Enter the GitHub Project URL" or just "Enter GitHub Project URL"

Comment (craigomatic) [reply samples/apps/github-qna-webapp-react/src/components/GitHubRepoSelection.tsx]: Agreed - made the heading consistent with the menu text:

image

File: samples/skills/QASkill/MemoryQuery/skprompt.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Someone wants to know the following about a GitHub repository:
```
[the remark this answers is not in the document]
Comment (lemillermicrosoft) [reply samples/skills/QASkill/MemoryQuery/skprompt.txt:1]: Yeah, it looks like that project is already defining a skill and functions. Can we put this there?

File: samples/dotnet/github-skills/GitHubSkill.cs [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+";
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GitHubSkill"/> class.
+    /// </summary>
+    /// <param name="kernel">Kernel instance</param>
+    /// <param name="downloadSkill">Instance of WebFileDownloadSkill used to download web files</param>
+    /// <param name="documentSkill">Instance of DocumentSkill used to read files</param>
+    /// <param name="logger">Optional logger</param>
+    public GitHubSkill(IKernel kernel, WebFileDownloadSkill downloadSkill, DocumentSkill documentSkill, ILogger<GitHubSkill>? logger = null)
```
[the remark this answers is not in the document]
Comment (shawncal) [reply samples/dotnet/github-skills/GitHubSkill.cs:89]: Ah, darn. Looks like we crossed streams... I just merged this PR into the "memory-sample" branch. Will open a new PR to main. @lemillermicrosoft can you repost these comments on that new PR? (stay tuned...)

File: samples/dotnet/github-skills/GitHubSkillsExample.csproj [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <AssemblyName>GitHubSkillsExample</AssemblyName>
+    <TargetFramework>netstandard2.1</TargetFramework>
```
Comment (lemillermicrosoft) [inline samples/dotnet/github-skills/GitHubSkillsExample.csproj:5]: Needs

```
  <Import Project="$(RepoRoot)/dotnet/Directory.Build.targets" />
  <Import Project="$(RepoRoot)/dotnet/Directory.Build.props" />
  ```

File: samples/dotnet/github-skills/GitHubSkill.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (shawncal) [reply samples/dotnet/github-skills/GitHubSkill.cs:89]: @lemillermicrosoft @craigomatic New PR: https://github.com/microsoft/semantic-kernel/pull/96

**Pull request merged**